.github/actions/setup/action.yml

```diff
@@ -0,0 +1,48 @@
+name: Setup
+description: Common setup steps used by our workflows
+runs:
+  using: composite
+  steps:
+    - name: Setup pnpm
+      uses: pnpm/action-setup@v2
+      with:
+        version: 8.6.5
+
+    - name: Setup node 16.x
+      uses: actions/setup-node@v3
+      with:
+        node-version-file: .nvmrc
+        registry-url: https://registry.npmjs.org
+        cache: pnpm
+
+    - name: Get foundry version
+      id: foundry-version
+      run: echo ::set-output name=version::$(cat .foundryrc)
+
+    - name: Setup foundry
+      uses: foundry-rs/foundry-toolchain@v1
```

Comment:
Is this going to cache? Its very slow to build foundry

Reply:
It's pretty fast everywhere else I used it. I think it's just pulling the image not building from source

Reply:
Foundry only makes a certain number of images available, as new ones are published it deletes old ones. I am pretty sure this will build from source given we are not using a super recent version

Reply:
Oh wait I think I did add the versioning though nvm one sec

Reply:
Ok confirmed the action is smart enough to cache https://github.com/foundry-rs/foundry-toolchain/blob/master/action.yml#L9 .